Sources/Client/Utils/InternetConnection.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  InternetConnection.swift
+//  StreamChatCore
+//
+//  Created by Alexey Bukhtin on 02/07/2019.
+//  Copyright © 2019 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+import Reachability
+
+/// The Internect connection manager.
+public final class InternetConnection {
+    
+    /// The Internet connection reachability.
+    public enum State {
+        /// Notification of an Internet connection has not begun.
+        /// Basically, this is the offline mode.
+        case none
```

Comment:
Now that I know the context, I can provide more useful suggestion: I'd say `none` should be `notObserving` or `unknown` because in this state, we don't know if we have internet or now

Reply:
That definitely means not observing.

Reply:
I'll rename it

Reply:
I'd prefer `unknown` but `notObserving` is good too